code/hud/hudshield.cpp

```diff
@@ -569,21 +567,28 @@ void HudGaugeShield::showShields(const object *objp, int mode, bool config)
 		sp = &Ships[objp->instance];
 		sip = &Ship_info[sp->ship_info_index];
 	} else {
-		SCP_string ship_name = "gtf myrmidon";
-
-		// Check if the mod specifies a differnt default
-		if (auto it = HC_hud_ships.find("default"); it != HC_hud_ships.end()) {
-			ship_name = it->second;
-		}
+		SCP_string ship_name;
 
 		// Now check if the current HUD specifies a different ship
 		if (SCP_vector_inbounds(HC_available_huds, HC_chosen_hud)) {
 			SCP_string hud = HC_available_huds[HC_chosen_hud].second;
-			if (auto it = HC_hud_ships.find(hud); it != HC_hud_ships.end()) {
-				ship_name = it->second;
+			if (auto it = HC_hud_shield_ships.find(hud); it != HC_hud_shield_ships.end()) {
+				ship_name = it->second[mode];
+			}
+		}
+
+		// If we don't have a specific setting then let's try using the global setting
+		if (ship_name.empty()) {
+			if (auto it = HC_hud_shield_ships.find("default"); it != HC_hud_shield_ships.end()) {
+				ship_name = it->second[mode];
 			}
 		}
 
+		// If we still don't have something then try to use the FS2 retail default
+		if (ship_name.empty()) {
+			ship_name = "gtf myrmidon";
+		}
```

Comment:
Is it plausible that using an empty string might be intentionally done on some mods to effectively hide this display?
Is there a reason why the order of operations / the behaviour was changed here in the first place?

Reply:
Hiding the shield gauge should be done by disabling the shield gauge, not by setting an invalid ship here, as that's suddenly implicit behavior instead of explicit.

The order was changed to make it easier to follow and easier to override. This version has multiple ways to override the generic setting and I felt the complexity was more clear this way.